californium-core/src/test/java/org/eclipse/californium/core/test/ClientSynchronousTest.java

```diff
@@ -150,6 +153,34 @@ public void testSynchronousCall() throws Exception {
 		Assert.assertEquals(5, notifications.get());
 		Assert.assertFalse(failed);
 	}
+
+	@Test
+	public void testAdvancedUsesTypeFromRequest() throws Exception {
+		String uri = "coap://localhost:"+serverPort+"/"+TARGET;
+		CoapClient client = new CoapClient(uri).useExecutor();
+
+		// Set NONs but expecting CONs as specified in request
+		client.useNONs();
+
+		Request request = new Request(Code.GET, Type.CON);
+
+		CoapResponse resp = client.advanced(request);
+
+		Assert.assertEquals(CONTENT_1, resp.getResponseText());
```

Comment:
Yes, thanks, sure we should...
I'll add the asserts but here is another one issue, I'll point to it. Need advice, please.